sklearn/_fast_gradient_boosting/binning.py

```diff
@@ -0,0 +1,140 @@
+"""
+This module contains the BinMapper class.
+
+BinMapper is used for mapping a real-valued dataset into integer-valued bins.
+Bin thresholds are computed with the quantiles so that each bin contains
+approximately the same number of samples.
+"""
+# Author: Nicolas Hug
+
+import numpy as np
+
+from ..utils import check_random_state, check_array
+from ..base import BaseEstimator, TransformerMixin
+from ..utils.validation import check_is_fitted
+from ._binning import _map_to_bins
+from .types import X_DTYPE, X_BINNED_DTYPE
+
+
+def _find_binning_thresholds(data, max_bins=256, subsample=int(2e5),
+                             random_state=None):
+    """Extract feature-wise quantiles from numerical data.
+
+    Return
```

Comment:
`Parameters` section too, please? :P (i.e. looking at the signature, it's not clear what `subsample`'s going to do).